### ISSUE TITLE
Minor regression with `Uri` affecting equality

### ISSUE THREAD

**Issue by gypsydave5** [http4k/http4k#189]
Noticed this after upgrading from ~1.27 this morning:

```kotlin
        val path = "path"
        val uri1 = Uri.of(path)
        val uri2 = Uri.of(uri1.toString())

        println(uri1.path)
        println(uri2.path)
        assertEqual(uri1, uri2)
```

This used to work OK. The representation of `path` in each `Uri` is different; `uri1` is now `"path"` and `uri2` is `"/path"` and so they fail object equality.

Not sure about the best way to approach fixing so opening issue rather than PR.

**Comment by daviddenton** [comment]
@gypsydave5 Thanks for the report. We're also considering options on the best way to fix this. There are a couple but we need to be careful not to cause random breaks! :)

**Comment by daviddenton** [comment]
Fixed in 3.39.3 🙃
